doc/embed_compiled_native.md

```diff
@@ -106,3 +110,127 @@ gcc -O3 -o nosandbox_with_staticlib ../src/nosandbox_main.c -L. -lstaticnosandbo
+
+extern int
+add(int a, int b);
+
+int
+main()
+{
+    printf("Nosandbox main: add 3 + 4 = %d\n", add(3, 4));
+}
+```
+
+So in this section, we will focus on the sandbox mode library.
+
+Here is the more detailed process of how to use the export APIs defined in `w2n_export.h` to interact with the wasm app(in the form of a library) in the host native binary:
+
+### Embed the wasm app library in the host native binary
+
+First, the wasm instance needs to be created using `wasm_instance_create()`, and check whether it creates successfully using `wasm_instance_is_created()`. If it fails or at the end of the host native binary, we need to destroy the wasm instance.
+
+```C
+    wasm_instance_create();
+    if (!wasm_instance_is_created()) {
+        os_printf("Create wasm instance failed.\n");
+        goto fail;
+    }
+
+    ...
+
+fail:
+    wasm_instance_destroy();
+    return ret;
+```
+
+Then you can use `wasm_get_export_apis()` and `wasm_get_export_api_num()` to get and call the wasm function by name, following similar procedures as below(a full example process can be found in `execute_func` of [wasm_application.c](wasm2native-vmlib/application/wasm_application.c)):
+
+```C
+    WASMExportApi *export_apis = wasm_get_export_apis();
+    WASMExportApi *export_func = NULL;
+    void (*invoke_native)(const void *, uint32 *, uint32 *);
+    uint32 export_api_num = wasm_get_export_api_num();
+
+    /* Lookup exported function */
+    for (i = 0; i < export_api_num; i++) {
+        if (!strcmp(export_apis[i].func_name, name)) {
+            export_func = &export_apis[i];
+            break;
+        }
+    }
+
+    /* from the export_func->signature, parse and prepare the arguments array(argv1) */
+    ...
+
+    /* lookup the quick aot entry, basically it will reinterpret the function ptr type that will match export_func->signature */
+    invoke_native = lookup_quick_aot_entry(export_func->signature);
+
+    /* call the wasm function, use the argv1 as both parameter and return array */
+    invoke_native(export_func->func_ptr, argv1, argv1);
+
+    /* parse and process the return value from the return array(argv1) */
+    ...
+```
+
+PS: during the wasm instance creation or calling the wasm function, you can use `wasm_get_exception()` and `wasm_get_exception_msg()` to check the exception and utilize the exception message.
+
+```C
+static void
+check_and_print_exception()
+{
+    int32 exce_id = wasm_get_exception();
+    const char*exce_msg;
+    if (exce_id) {
+        exce_msg = wasm_get_exception_msg();
+        os_printf("Exception: %s\n", exce_msg);
+    }
+    else {
+        os_printf("Exception: unknown error\n");
+    }
+}
+
+// check whether create wasm instance successfully
+int
+main(int argc, char *argv[])
+{
+    ...
+
+    if (!wasm_instance_is_created()) {
+        os_printf("Create wasm instance failed.\n");
+        check_and_print_exception();
+        goto fail;
+    }
+
+    ...
+    
+    if (!app_instance_func(func_name))
```

Comment:
don't call app_instance_func, it is also provided in wasm_application.c and suppose not to be linked in some scenarios.